apiserver/facades/client/spaces/shims.go

```diff
@@ -59,3 +62,92 @@ func (s *stateShim) SubnetByCIDR(cidr string) (networkingcommon.BackingSubnet, e
 	}
 	return networkingcommon.NewSubnetShim(result), nil
 }
+
+func (s *stateShim) SpaceByName(name string) (networkingcommon.BackingSpace, error) {
+	result, err := s.State.SpaceByName(name)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	space := networkingcommon.NewSpaceShim(result)
+	return space, nil
+}
+
+func (s *stateShim) RenameSpace(fromSpaceName, toName string) error {
+	var totalOps []txn.Op
+
+	constraintsOps, err := updateConstraints(s.State, fromSpaceName, toName)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	totalOps = append(totalOps, constraintsOps...)
+
+	settingsOps, err := getSettingsOps(s.State, fromSpaceName, toName)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	totalOps = append(totalOps, settingsOps...)
+
+	err = s.State.RenameSpace(fromSpaceName, toName, totalOps)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	return nil
+}
+
+func getSettingsOps(st *state.State, fromSpaceName, toName string) ([]txn.Op, error) {
```

Comment:
might want to put that (plus below) in the facade and the needed state access into the backing interface.
Interface methods:
- controllerConfig
- ModelConstraints

which would make mocking easier.